server/routes/mentions.js

```diff
@@ -0,0 +1,76 @@
+const router = require("express").Router();
+
+const requiresAuth = require("./middleware/requiresAuth");
+const { Mention, User } = require("../models");
+
+const callScraper = require("../scraper");
+
+// TODO test mentions in action
+
+router.get("/", requiresAuth, async (req, res) => {
+  const user = await User.findByPk(req.user.id);
+  // note that the getter method for n-m can accept options
+  // as if it were a regular query
+  const mentions = await user.getMentions();
+
+  res.json(mentions);
+});
+
+router.get("/:id", requiresAuth, async (req, res) => {
+  const mention = await Mention.findByPk(req.params.id);
+
+  res.json(mention);
+});
+
+router.post("/", requiresAuth, async (req, res) => {
+  // NOTE I think this is better served as just a test route in order to test the scraper
+  // This shouldn't be placed in the final PR as there will be another means to post new data
+  // into the database.
+  const user = await User.findByPk(req.user.id);
+
+  let mentions = callScraper(req.user.company, 100);
+
+  let result = [];
+  for (let m of mentions) {
+    let userMentions = await user.getMentions({
+      where: {
+        title: m.title,
+      },
+    });
+
+    // if this is new
+    if (!userMentions.length) {
+      [mention, isNew] = await Mention.findOrCreate({
+        where: {
+          title: m.title,
+        },
+        defaults: {
+          title: m.title,
+          platform: m.platform,
+          date: m.date,
+          content: m.content || "",
+          popularity: m.popularity,
+          imageUrl: m.image,
+        },
+      });
+      result.push(mention);
```

Comment:
I guess, in this line, only new mentions will be stored in `result` and be returned to frontend?
I think if we want to return mentions to frontend, we need to return all mentioned that we just scraped instead of only those that didn't exsit in DB.